src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -216,44 +206,42 @@ internal override async ValueTask<QuicStreamProvider> AcceptStreamAsync(Cancella
         internal override QuicStreamProvider OpenUnidirectionalStream()
         {
             ThrowIfDisposed();
-
-            return StreamOpen(QUIC_STREAM_OPEN_FLAG.UNIDIRECTIONAL);
+            return new MsQuicStream(_state.Handle, QUIC_STREAM_OPEN_FLAGS.UNIDIRECTIONAL);
         }
 
         internal override QuicStreamProvider OpenBidirectionalStream()
         {
             ThrowIfDisposed();
-
-            return StreamOpen(QUIC_STREAM_OPEN_FLAG.NONE);
+            return new MsQuicStream(_state.Handle, QUIC_STREAM_OPEN_FLAGS.NONE);
         }
 
         internal override long GetRemoteAvailableUnidirectionalStreamCount()
         {
-            return MsQuicParameterHelpers.GetUShortParam(MsQuicApi.Api, _ptr, (uint)QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.PEER_UNIDI_STREAM_COUNT);
+            return MsQuicParameterHelpers.GetUShortParam(MsQuicApi.Api, _state.Handle, QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.LOCAL_UNIDI_STREAM_COUNT);
         }
 
         internal override long GetRemoteAvailableBidirectionalStreamCount()
         {
-            return MsQuicParameterHelpers.GetUShortParam(MsQuicApi.Api, _ptr, (uint)QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.PEER_BIDI_STREAM_COUNT);
-        }
-
-        private void SetIdleTimeout(TimeSpan timeout)
-        {
-            MsQuicParameterHelpers.SetULongParam(MsQuicApi.Api, _ptr, (uint)QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.IDLE_TIMEOUT, (ulong)timeout.TotalMilliseconds);
+            return MsQuicParameterHelpers.GetUShortParam(MsQuicApi.Api, _state.Handle, QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.LOCAL_BIDI_STREAM_COUNT);
         }
 
         internal override ValueTask ConnectAsync(CancellationToken cancellationToken = default)
         {
             ThrowIfDisposed();
 
+            if (_configuration is null)
+            {
+                throw new Exception($"{nameof(ConnectAsync)} must not be called on a connection obtained from a listener.");
+            }
+
             (string address, int port) = _remoteEndPoint switch
             {
                 DnsEndPoint dnsEp => (dnsEp.Host, dnsEp.Port),
                 IPEndPoint ipEp => (ipEp.Address.ToString(), ipEp.Port),
                 _ => throw new Exception($"Unsupported remote endpoint type '{_remoteEndPoint.GetType()}'.")
             };
 
-            // values taken from https://github.com/microsoft/msquic/blob/main/docs/api/ConnectionStart.md
+            // TODO: MsQuic will use system constants, so we should use the Socket PAL to translate these.
```

Comment:
//
// Defines match windows values.
//
#define QUIC_ADDRESS_FAMILY_UNSPEC 0
#define QUIC_ADDRESS_FAMILY_INET 2
#define QUIC_ADDRESS_FAMILY_INET6 23

typedef union QUIC_ADDR {
    struct sockaddr Ip;
    struct sockaddr_in Ipv4;
    struct sockaddr_in6 Ipv6;
} QUIC_ADDR;

since they decided to follow Window for their enums, we should be good here. (since our enum follow Windows as well). QUIC_ADDRESS_FAMILY_INET is no brainer but QUIC_ADDRESS_FAMILY_INET6 would break if we pass in OS INET6 e.g. use something like `TryConvertAddressFamilyPalToPlatform()`

I would simply check the types and pass in `(int)_remoteEndPoint.AddressFamily` (and add note)

Reply:
Are we sure this is the case? I added this comment because it didn't seem like msquic was doing any sort of translation here.

Reply:
It is hidden in msquic platform. places like:

https://github.com/microsoft/msquic/blob/a0dd95474c40ea3f675b6a58083b1149c9b0b19d/src/platform/datapath_epoll.c#L832-L838

Windows needs no translation.

Reply:
Yes. We (@thadhouse) did the work to expose a platform independent interface here (by standardizing on Windows definitions).

Reply:
Added `QUIC_ADDRESS_FAMILY` enum and changed the usages.